EXILED/Exiled.API/Features/Toys/Speaker.cs

```diff
@@ -80,5 +104,187 @@ public float MinDistance
+                    yield break;
+                }
+
+                Log.Info($"Playing OGG file with Sample Rate: {sampleRate}, Channels: {channels}");
+
+                while ((streamBuffer.Count < frameSize * 2 && !stopPlayback) || Round.IsEnded)
+                {
+                    int samplesRead = vorbisReader.ReadSamples(readBuffer, 0, readBuffer.Length);
+                    if (samplesRead <= 0)
+                        break;
+
+                    foreach (float sample in readBuffer.Take(samplesRead))
+                        streamBuffer.Enqueue(sample);
+
+                    while (!stopPlayback && streamBuffer.Count > 0)
+                    {
+                        if (Timing.LocalTime < nextPlaybackTime)
+                        {
+                            yield return Timing.WaitForOneFrame;
+                            continue;
+                        }
+
+                        for (int i = 0; i < frameSize && streamBuffer.Count > 0; i++)
+                            sendBuffer[i] = streamBuffer.Dequeue();
+
+                        int dataLen = encoder.Encode(sendBuffer, encodedBuffer);
+                        AudioMessage audioMessage = new(ControllerID, encodedBuffer, dataLen);
+                        audioMessage.SendToAuthenticated();
+
+                        Log.Debug($"Sent {dataLen} bytes of encoded audio.");
+
+                        nextPlaybackTime += playbackInterval;
+
+                        if (streamBuffer.Count < frameSize && !vorbisReader.IsEndOfStream)
+                        {
+                            samplesRead = vorbisReader.ReadSamples(readBuffer, 0, readBuffer.Length);
+                            foreach (float sample in readBuffer.Take(samplesRead))
+                                streamBuffer.Enqueue(sample);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                Log.Error($"Unsupported file format: {fileExtension}");
+                yield break;
+            }
+
+            Log.Info("Playback completed.");
+            IsPlaying = false;
+            if (destroyAfter)
+                Destroy();
+        }
+
+        // Naudio Support :|
+        // private int ConvertBytesToFloats(byte[] byteBuffer, float[] floatBuffer, int bytesRead, WaveFormat waveFormat)
+        // {
+        //     int bytesPerSample = waveFormat.BitsPerSample / 8;
+        //     int sampleCount = bytesRead / bytesPerSample;
+        //
+        //     for (int i = 0; i < sampleCount; i++)
+        //     {
+        //         // Assuming 16-bit PCM data
+        //         short sample = BitConverter.ToInt16(byteBuffer, i * bytesPerSample);
+        //         floatBuffer[i] = sample / 32768f; // Normalize to -1.0 to 1.0 range
+        //     }
+        //
+        //     return sampleCount;
+        // }
```

Comment:
Are you going to add support on this?

Reply:
No it was just in case NVorbis wasn't accepted for the license or any related issue

Reply:
So what about this?

Reply:
Ok i think we can solve it